### PR TITLE
Split all-packages.nix

### PR DESCRIPTION
**Pull request by nbp**
This patch is moves the header & footer of `all-packages.nix` into a separated file named `all-packages-wrapper.nix`.  To do so, I added a new list of arguments to the `all-packages.nix` which is the explicit list of dependencies, and that we can reduce over time by making it part of the list of paclkages. (`system`, `bootStdenv`, `noSysDir`, …)

Later we should change the way we handle self-references (`self` & `pkgs`) such that we can follow the same scheme as done in haskellPackages (`self` & `super`).

These modifications are made to simplify the way we can add additional logic to our list of packages.

Note: This patch does not change the sha1 of derivations.
Note: This patch fix the indentation issue of `all-packages.nix`.
cc @peti @domenkozar @edolstra

### CODE REVIEW

**Comment by jagajaga** [comment]
cc @edolstra

**Comment by peti** [comment]
The error https://travis-ci.org/NixOS/nixpkgs/builds/84793391#L116 from travis looks genuine. It would be great to get that fixed. Otherwise, I'm in favor of merging this patch: :+1:. Does anyone consider this re-factoring controversial?

**Comment by nbp** [comment]
I rebased this work on top of another pull request that is cleaning the logic of override packages and stdenv.  I am currently working on a security update branch for the NixCon, which is based on a similar modification.

I will get back to this, hopefully before NixCon ;)

File: pkgs/stdenv/darwin/default.nix [hunk 1]
```diff
@@ -1,11 +1,11 @@
 { system      ? builtins.currentSystem
-, allPackages ? import ../../top-level/all-packages.nix
+, allPackages ? import ../../top-level/all-packages-wrapper.nix
 , platform    ? null
 , config      ? {}
 }:
 
 rec {
-  allPackages = import ../../top-level/all-packages.nix;
+  allPackages = import ../../top-level/all-packages-wrapper.nix;
```
Comment (edolstra) [inline pkgs/stdenv/darwin/default.nix]: Why not refer to `../../..` (i.e. the top-level `default.nix`)?

File: pkgs/top-level/make-tarball.nix [hunk 2]
```diff
@@ -49,13 +49,13 @@ releaseTools.sourceTarball rec {
         exit 1
     fi
 
-    # Check that all-packages.nix evaluates on a number of platforms.
+    # Check that all-packages-wrapper.nix evaluates on a number of platforms.
     for platform in i686-linux x86_64-linux x86_64-darwin; do
-        header "checking pkgs/top-level/all-packages.nix on $platform"
-        NIXPKGS_ALLOW_BROKEN=1 nix-env -f pkgs/top-level/all-packages.nix \
+        header "checking pkgs/top-level/all-packages-wrapper.nix on $platform"
+        NIXPKGS_ALLOW_BROKEN=1 nix-env -f pkgs/top-level/all-packages-wrapper.nix \
```
Comment (edolstra) [inline pkgs/top-level/make-tarball.nix]: Similarly, `nix-env -f .` should work just as well.

File: pkgs/stdenv/darwin/default.nix (hunk 1 above)
Comment (nbp) [inline pkgs/stdenv/darwin/default.nix]: Good point. I will do that.

**Comment by nbp** [comment]
This work is now addressed as part of #10851.

**Pull request closed without merging**